### DISCUSSION TITLE
npm uninstall <pkg> -g isn't doing what it's supposed to do

### DISCUSSION THREAD

**Discussion by ChristianSchieder** [coreybutler/nvm-windows#1101] [General] [answered]
So to explain a little, im using nvm inside a virtual machine running on hyper-v manager.

I have trouble uninstalling a package which got installed from a private registry.
using:  `npm install --registry=path-to-registry` <pkg-name> -g

The issue seems to occur after pkg installation from the private registry, which leads me clueless what the reason for that might be, nvm issue, npm issue, package issue or a combination of all 3.

Why do i think it has to do with the package beeing from a private registry?
Because if i install packages from the npm registry globaly the removal is working correctly.

If i have this one package installed from the private registry and others from npm registry globaly, the problem affects the other packages as well, and npm uninstall of any globaly installed package won't work anymore.

running `nvm debug` will give me this output:

cschieder is not using admin or elevated rights.
Error getting console handle: Der Vorgang wurde erfolgreich beendet.
Windows Version: 10.0 (Build 19045)
 
Windows Developer Mode: UNKNOWN (user cannot read registry)
 
NVM4W Version:      1.1.12
NVM4W Path:         C:\Users\cschieder\AppData\Roaming\nvm\nvm.exe     
NVM4W Settings:     C:\Users\cschieder\AppData\Roaming\nvm\settings.txt
NVM_HOME:           C:\Users\cschieder\AppData\Roaming\nvm
NVM_SYMLINK:        C:\Program Files\nodejs
Node Installations: C:\Users\cschieder\AppData\Roaming\nvm
 
Total Node.js Versions: 1
Active Node.js Version: v18.18.0
 
PROBLEMS DETECTED
-----------------
Connection type cannot be determined: dial tcp [::1]:80: connectex: Es konnte keine Verbindung hergestellt werden, da der Zielcomputer die Verbindung verweigerte.

I couldn't find much on the detected problem itself, just one issue where it was mentioned but in a other context.

Any help would be apreciated, and if i need to provide some additional infos let me know.

**Comment by coreybutler** [comment]
This wouldn't be an NVM4W issue. Once npm is installed, NVM4W is completely out of the picture. However; the nvm debug output may provide a clue. My German is pretty rusty, but I believe the error is telling you it cannot connect to the network. This could be caused by the registry denying access, a firewall on your computer denying access, or a proxy (if enabled) denying access.

**Reply by ChristianSchieder** [reply]
Since NVM detects this kind of error, do you maybe have a clue, how this error state gets evaluated?
Is it some NPM diagnostic tool or even a custom util of NVM, or neither of both and it's some machine state?
Just want to narrow down if i might need to describe this as issue in the NPM repository, or ask for some troubleshooting help elsewhere.

Thanks so far!

**Reply by coreybutler** [reply, marked as answer]
The error you're receiving comes from the [IPv6 check](https://github.com/coreybutler/nvm-windows/blob/master/src/nvm.go#L1156-L1158). It's a generic check that [attempts to contact localhost](https://github.com/coreybutler/nvm-windows/blob/master/src/web/web.go#L70-L81) via TCP (essentially a ping) on the IPv6 local address (`[::1]`). It looks like your system is configured to use IPv6, but something is wrong. In particular, your system seems to be hitting [this line](https://github.com/coreybutler/nvm-windows/blob/master/src/web/web.go#L77) of code, which is an Operating System level error (i.e. unknown to NVM4W). This is Go's generic network dialer, the part that is responsible for initiating any kind of network connection.

If your system is also configured to use IPv4 and you're successfully able to `npm install` other public modules, then your private registry connection is the problem. I don't know how npm represents itself when making HTTP requests from an IPv6 address, so I would refer to the npm team to help address that. The simplest way to test this would be to disable IPv6 to see if it works.

Again, I think this is beyond the scope of NVM4W, but the checks NVM4W runs suggest the issue has something to do with the IPv6 network.

**Reply by ChristianSchieder** [reply]
Even if it's beyond NVM responsibility, thanks a lot for the detailed answer and the hints, i highly apreciate it!
